en/contribute/plugin.md

```diff
@@ -1,9 +1,868 @@
+<style>
+.markdown-body h4{
+  font-size: 1.2em;
+  padding-top: 1em;
+}
+.markdown-body h3{
+  font-size: 1.45em;
+}
+</style>
+
 # How To Write A Plugin
 
-@TODO
+>**Use [`svrx-create-plugin`](https://github.com/svrxjs/svrx-create-plugin)** to help you create plugins more easily
+
+### First Plugin
+
+Let's create our first plugin —— [`svrx-plugin-hello-world`](https://github.com/svrxjs/svrx-plugin-hello-world)
+
+### File Structure
+
+```js
+└── svrx-plugin-hello-world
+  ├── client.js
+  ├── index.js
+  └── package.json
+```
+
+> _Only `package.json` and `index.js`_ are required
+
+- `package.json`
+
+```js
+{
+  "name" : "svrx-plugin-hello-world",
+  "engines": {
+    "svrx" : "0.0.x"
+  }
+}
+```
+
+Only two fields are required by `package.json`
+
+- `name`: package name must be a string beginning with `svrx-plugin`, to help `svrx` find it in npm.
+- `engines.svrx`: Define the runnable svrx version of this plugin，svrx will automatically load the latest matching plugin
+  **engines.svrx** can be a valid [semver](https://www.npmjs.com/package/semver)，like `0.1.x` or `~0.1`
+
+#### - `index.js`
+
+```js
+module.exports = {
+  configSchema: {
+    user: {
+      type: 'string',
+      default: 'svrx',
+      description: 'username for hello world'
+    }
+  },
+
+  assets: {
+    script: ['./client.js']
+  },
+
+  hooks: {
+    async onCreate({ logger, config }) {
+      logger.log(`Hello ${config.get('user')} from server`);
+    }
+  }
+};
+```
+
+Where
+
+1. `configSchema` Plugin param definition based json-schema ，checkout [json-schema](#schema) for more detail.
+   Here we just set a `user` field, it is a string
+2. `assets`: client resource setting，they will be automatically injected into the page
+
+   - `style`: css resource injection
+   - `script`: script resource injection
+
+     > All resources will be merged into one link
+
+3. `hook.onCreate`: a function invoke when plugin been creating, we can control it by injected service
```

Comment:
`a function invoke when plugin been creating, we can control it by injected service`

->

`a function invoked during plugin initialization, we can overwrite it by injected services`